src/compile/axis.ts

```diff
@@ -92,8 +92,14 @@ export function parseAxis(channel: Channel, model: Model): VgAxis {
   };
 
   // format mixins (add format and formatType)
-  extend(def, formatMixins(model, channel, model.axis(channel).format));
-
+  // In case we need to get the fielddef axis for X2 or Y2
+  // if X or Y doesn't exist in the model. Use targetChannel
+  // to access X2 or Y2.
+  let targetChannel = channel;
+  if (!model.has(channel)) {
+    targetChannel = channel === X ? X2 : Y2;
+  }
+  extend(def, formatMixins(model, model.fieldDef(targetChannel), axis.format, axis.shortTimeLabels));
```

Comment:
Check if we really need other properties in `fieldDef`  or we simply need `fieldDef.axis`.  If it's the latter case, remove this `targetChannel` thing here because we won't have `axis` in `x2` / `y2`.